asset-transfer-basic/chaincode-java/src/main/java/org/hyperledger/fabric/samples/assetTransfer/AssetTransfer.java

```diff
@@ -0,0 +1,238 @@
+                contact = @Contact(
+                        email = "a.transfer@example.com",
+                        name = "Adrian Transfer",
+                        url = "https://hyperledger.example.com")))
+@Default
+public final class AssetTransfer implements ContractInterface {
+
+    private final Genson genson = new Genson();
+
+    private enum AssetTransferErrors {
+        ASSET_NOT_FOUND,
+        ASSET_ALREADY_EXISTS
+    }
+
+    /**
+     * Creates some initial Assets on the ledger.
+     *
+     * @param ctx the transaction context
+     */
+    @Transaction()
+    public void InitLedger(final Context ctx) {
+        ChaincodeStub stub = ctx.getStub();
+
+        CreateAsset(ctx, "asset1", "blue", 5, "Tomoko", 300);
+        CreateAsset(ctx, "asset2", "red", 5, "Brad", 400);
+        CreateAsset(ctx, "asset3", "green", 10, "Jin Soo", 500);
+        CreateAsset(ctx, "asset4", "yellow", 10, "Max", 600);
+        CreateAsset(ctx, "asset5", "black", 15, "Adrian", 700);
+        CreateAsset(ctx, "asset6", "white", 15, "Michel", 700);
+
+    }
+
+    /**
+     * Creates a new asset on the ledger.
+     *
+     * @param ctx the transaction context
+     * @param assetID the assetID of the new asset
+     * @param color the color of the new asset
+     * @param size the size for the new asset
+     * @param owner the owner of the new asset
+     * @param appraisedValue the appraisedValue of the new asset
+     * @return the created Asset
+     */
+    @Transaction()
+    public Asset CreateAsset(final Context ctx, final String assetID, final String color, final int size,
+        final String owner, final int appraisedValue) {
+        ChaincodeStub stub = ctx.getStub();
+
+        if (AssetExists(ctx, assetID)) {
+            String errorMessage = String.format("Asset %s already exists", assetID);
+            System.out.println(errorMessage);
+            throw new ChaincodeException(errorMessage, AssetTransferErrors.ASSET_ALREADY_EXISTS.toString());
+        }
+
+        Asset asset = new Asset(assetID, color, size, owner, appraisedValue);
+        String assetState = genson.serialize(asset);
+        stub.putStringState(assetID, assetState);
+
+        return asset;
+    }
+
+    /**
+     * Retrieves an asset with the specified ID from the ledger.
+     *
+     * @param ctx the transaction context
+     * @param assetID the assetID
+     * @return the Asset found on the ledger if there was one
+     */
+    @Transaction()
+    public Asset ReadAsset(final Context ctx, final String assetID) {
+        ChaincodeStub stub = ctx.getStub();
+
+        if (!AssetExists(ctx, assetID)) {
+            String errorMessage = String.format("Asset %s does not exist", assetID);
+            System.out.println(errorMessage);
+            throw new ChaincodeException(errorMessage, AssetTransferErrors.ASSET_NOT_FOUND.toString());
+        }
+
+        String assetState = stub.getStringState(assetID);
```

Comment:
you can avoid AssetExists call in line 110 , by just checking if getStringState found the asset

```
if (assetState == null ||  isEmpty() )
   throw new ChaincodeException as above
```